### DISCUSSION TITLE
onSessionActive not called with more than one connections

### DISCUSSION THREAD

**Discussion by pglamm** [eclipse-milo/milo#1131] [General]
I have been using the sdk-client for some time with great success. Now I am setting up a system, where I need to connect with more than one connection. I have wrapped the sdk inside my own class, so all handles, callbacks etc. are unique for each connection. I use the ProSYS OPC UA Simulation server as test, and have been using the same wrapper code for both the single connect system and now the multi connect.

What I see is that my `onSessionActive` is called for the first connect, but not for the second. I tried to switch second connection to connect to the DigitalPetri test server, but I get the same result. I subscribe for data immediately after 

Some very curious observations:
- when I restart the server, `onSessionActive` is called for both connections when the server has started again
- when i set the log level of `org.eclipse.milo.opcua.sdk.client.SessionFsm` to `TRACE`, I do get `onSessionActive` for both connections

Both connections seem to work, nodes are subscribed to, and changes on server are sent to my application. It is just the session active notification on the second connection during startup that is missing. Any idea? 

This is my essential connect code (error handling etc. excluded):

```java
OpcUaClient = opcUaClient = createOpcUaClient();
opcUaClient.connect().get();
opcUaClient.addSessionActivityListener(sessionActivityListener);
opcUaClient.getSubscriptionManager().addSubscriptionListener(subscriptionListener);
// Create subscriptions here
```

The `OpcUaClient` is created in the following way

```java
private OpcUaClient createOpcUaClient() {
  var securityPolicy = SecurityPolicy.Basic256Sha256;
  return OpcUaClient.create(
      opc.tcp://localhost:53530/OPCUA/SimulationServer,
        endpoints -> endpoints.stream()
            .filter(e -> securityPolicy.getUri().equals(e.getSecurityPolicyUri()))
            .findFirst(),
        configBuilder -> configBuilder
            .setApplicationName(LocalizedText.english("My App 1")) // "2" for second connection
            .setApplicationUri(urn:app1.xxx.dk:opcua) // "2" for second connection
            .setKeyPair(keystore.getClientKeyPair())
            .setCertificate(keystore.getClientCertificate())
            .setCertificateChain(keystore.getClientCertificateChain())
            .setCertificateValidator(new DefaultClientCertificateValidator(new DefaultTrustListManager("path-to-pki-dir")))
            .setIdentityProvider(new AnonymousProvider())
            .setConnectTimeout(Unsigned.uint(5000))
            .setSessionTimeout(Unsigned.uint(30000))
            .setRequestTimeout(Unsigned.uint(2000))
            .setKeepAliveInterval(Unsigned.uint(5000))
            .setKeepAliveTimeout(Unsigned.uint(5000))
            .setKeepAliveFailuresAllowed(Unsigned.uint(1))
            .build());
}
```

**Comment by kevinherron** [comment]
You should be calling `addSessionActivityListener()` before you call `connect()`.

If that's not the issue, I'll need a self-contained example that can reproduce it. We use many simultaneous client instances and don't seem to have any issue.

**Reply by pglamm** [reply]
Thank you @kevinherron, moving the call to before `connect()` did the trick.
